shadow-dom/focus/resources/shadow-utils.js

```diff
@@ -69,3 +70,14 @@ function navigateFocusForward() {
   return test_driver.send_keys(document.body, "\ue004");
 }
 
+function assertFocusOrder(expectedOrder) {
```

Comment:
The `return new Promise` isn't needed here. Just make the whole function async, and then put the loop body inside the function:

```js
async function assertFocusOrder(expectedOrder) {
  let shadowRoot = document.getElementById("host").shadowRoot;
  for (let el of expectedOrder) {
    await navigateFocusForward();
    let focused = shadowRoot.activeElement ? shadowRoot.activeElement : document.activeElement;
    assert_equals(focused, el);
  }
}
```

Reply:
Oh awesome, thanks :D